File: API/Helpers/Pagination.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class Pagination<T> where T : class
    {
        #region 6.65 Add class and properties for pagination ->iGenericRepository
        public Pagination(int pageIndex, int pageSize, int count, IReadOnlyList<T> data)
        {
            PageIndex = pageIndex;
            PageSize = pageSize;
            Count = count;
            Data = data;
        }
        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public int Count { get; set; }
        public IReadOnlyList<T> Data { get; set; }
        #endregion
    }
}

File: client/src/app/shared/models/pagination.ts
import { IProduct } from "./product";

//#region 8.84.2 add pagination interface and properties -> app.ts
export interface IPagination {
    pageIndex: number;
    pageSize: number;
    count: number;
    data: IProduct[];
}
// 21.284 Adding properties in order to cache pagination -> shop.service.ts
export class Pagination implements IPagination {
    pageIndex: number;
    pageSize: number;
    count: number;
    data: IProduct[] = [];
}
//#endregion